packages/stockflux-components/src/components/preview-window/PreviewWindow.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { StockFluxHooks } from 'stockflux-core';
+
+export default ({
+  htmlfile = 'transparent-window.html',
+  children,
+  position = { left: 100, top: 100 },
+  size = { width: 300, height: 300 },
+  display
+}) => {
+  const TRANSPARENT_WINDOW_NAME = 'transparent_window';
+  const TRANSPARENT_CSS_PATCH = 'transparentWindow.css';
+
+  const windowStatus = {
+    showing: 'SHOWING',
+    hiding: 'HIDING',
+    hide: 'HIDE',
+    hidden: 'HIDDEN'
+  };
+
+  const childWindow = StockFluxHooks.useChildWindow(
+    TRANSPARENT_WINDOW_NAME,
+    document,
+    TRANSPARENT_CSS_PATCH
+  );
+
+  const { window, launch } = childWindow;
+
+  const launchChildWindow = useCallback(
+    () =>
+      launch({
+        name: TRANSPARENT_WINDOW_NAME,
+        waitForPageLoad: true,
+        opacity: 0.0,
+        autoShow: false,
+        showTaskbarIcon: false,
+        defaultHeight: size.height,
+        defaultWidth: size.width,
+        url: htmlfile,
+        frame: false,
+        resizable: false,
+        alwaysOnTop: true
+      }),
+    [launch]
```

Comment:
I know`react-hooks/exhaustive-deps` is a strict rule, but sometimes it does help to pinpoint dependencies. Shouldn't these be included in the dependency array: 'htmlfile', 'size.height', and 'size.width'? It's ok if they should not, just double checking.

Reply:
I have added `// eslint-disable-next-line react-hooks/exhaustive-deps` to this and the other ones in this file. I have checked through dependencies and they are not needed for the hooks in question.